PROTOCOL.md

```diff
@@ -0,0 +1,347 @@
+associates with the job to run. For example, the operator might include the
+`targets` in a `Dockerfile` and then build it with `docker build`.
+4. The `operator` schedules the job to run. For example, this might include
+configuring the aforementioned container to run under `kubernetes`.
+
+All remaining steps are identical to the `Remote Push` model.
+
+### Phase #2: Provisioning
+
+The provisioning phase is identical to the `Remote Push` model, with the
+exception that the `keep` will connect to the `targetsrv` instead of the `enarx
+run` instance. This is because the `target` contains a different `uri`.
+Further, the `targetsrv` will use the `pubₜ` key to uniquely identify which set
+of `items` to deploy. Other than these differences, this process is identical
+to the `Remote Push` model.
+
+## Model #3: Multi-Party Computation
+
+As an astute reader, you've probably noticed that we are passing `targets` into
+the `keep` instantiation process but that this has, until now, only had a
+single `target` configured. Under the multi-party computation model, we are
+going to assembly a composite `keep` containing `items` from multiple different
+parties. This model builds on the previous models by allowing the `keep` to
+contact multiple `targetsrv` instances and merging each of their `manifest`s
+into one canonical manifest (`canon`). Let's survey the whole process. Note
+that all the details in this process are the same as previous models.
+Therefore, we will elide many details to make it easier to reason about. This
+does not imply these details are dropped under this model.
+
+```mermaid
+graph RL
+    subgraph parties
+        A(operator)
+        B(targetsrv)
+    end
+
+    subgraph host
+        C(enarx run)
+        D(enarx srv)
+        E(keep)
+
+	    style E fill:#0f0,stroke:#333,stroke-width:4px
+    end
+
+    A -. "1. enarx push" .-> B
+    B -. "2. pubₜ" .-> A
+    A -. "3. (uri, pubₜ)" .-> C
+
+    C -.-> D
+    D -.-> E
+
+    E -- "4. attest" --> B
+    B -- "5. manifest" --> E
+    E -. "6. canonicalize" .-> E
+    E -- "7. canon, [eitemhash]" --> B
+    B -- "8. [items]" --> E
+```
+
+1. Each party in the computation uses an agreed upon `enarx` build. Each party
+does an `enarx push` to store their `items` in their own `targetsrv`.
+2. Storing `items` in a `targetsrv` produces a public key (`pubₜ`) which
+references them.
+3. Each party submits its `target` configuration which are combined into
+`targets`. The `targets` input is supplied to the `enarx run` command, which
+can be executed either manually or by a scheduler. A `keep` is instantiated
+like normal.
+4. The `keep` attests to each `targetsrv`.
+5. Each `targetsrv` reports the manifest for all the items it is willing to
+deploy.
+6. The canonicalized manifest (`canon`) is produced by merging the `manifest`
+from each `targetsrv`.
+7. This step is where the magic happens. The `keep` requests the required
```

Comment:
Honestly, not as magic as the other step I noted earlier.  Also, remove, replace with "This step is where this process is different from the other approaches" or similar?